sdk/daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV2.scala

```diff
@@ -160,6 +168,18 @@ private[archive] class DecodeV2(minor: LV.Minor) {
     Work.run(decodeInternedTypes(env, lfPackage))
   }
 
+  private def decodeInternedKinds(
+      env: Env,
+      lfPackage: PLF.Package,
+  ): Work[IndexedSeq[Kind]] = Ret {
+    val lfKinds = lfPackage.getInternedKindsList
```

Comment:
```suggestion
    val lfKinds = lfPackage.getInternedKindsList
    (!lfTypes.isEmpty)
      assertSince(Features.kindInterning, "interned types table")
```

with a test for that.